backend/src/main/scala/bloop/CompilerCache.scala

```diff
@@ -211,13 +211,19 @@ final class CompilerCache(
             val newClasspathValue = newInvalidatedEntry.toFile + File.pathSeparator + classpathValue
             options(classpathValueIndex) = newClasspathValue
 
+            output.getSingleOutputAsPath match {
+              case p if p.isPresent => java.nio.file.Files.createDirectories(p.get)
+              case _ =>
+            }
```

Comment:
```suggestion
            output.getSingleOutputAsPath.ifPresent{p => 
              java.nio.file.Files.createDirectories(p)
              ()
            }
```
is it better than `p.isPresent` and `p.get` or not?

Reply:
Both don't look great to be honest and this follows the style in Zinc/sbt

Reply:
I personally don't like pattern matching when the side effect is executed only for one branch, it kind of blurs it, `asScala` + `foreach` would be great.

Reply:
changed to `ifPresent` here and in other place (LocalCompiler) also.